test/common_nn.py

```diff
@@ -390,6 +390,38 @@ def softmarginloss_reference(input, target, size_average=True, reduce=True):
     return output
 
 
+def _multimarginloss_reference(input, target_idx, p, margin, weight):
+    if weight is None:
+        weight = input.new(len(input)).fill_(1)
+
+    output = 0
+    for i in range(0, len(input)):
+        if i != target_idx:
+            output += max(0, weight[target_idx] * (margin - input[target_idx] + input[i]) ** p)
+    return output
+
+
+def multimarginloss_reference(input, target, p=1, margin=1, weight=None, size_average=True,
+                              reduce=True):
+    if input.dim() == 1:
+        n = 1
+        dim = input.size(0)
+        output = 0
+        return _multimarginloss_reference(input, target[0], p, margin, weight) / dim
+    else:
+        n = input.size(0)
+        dim = input.size(1)
+        output = input.new(n).zero_()
+        for x in range(0, n):
+            output[x] = _multimarginloss_reference(input[x], target[x], p, margin, weight)
+
+    if reduce and size_average:
+        return output.mean() / dim
+    elif reduce:
+        return output.sum() / dim
+    return output / dim
```

Comment:
Assuming this is correct, it seems that the equation in doc below is wrong. Should it divide by `x.size(1)` instead of `x.size(0)` in the equation.

Also, could you indent this block into the `else` branch. It makes it clearer.

Reply:
I think that in general, the equations in the doc refer to individual samples in a batch. (as in, x in the equation actually refers to a row of the input). I agree that it's a bit confusing, especially in this case, so I'll try to make it clearer.